Automattic/wp-calypso
# P2 signup: add 'join workspace' step

#### Changes proposed in this Pull Request

* Adds a new step `p2-join-workspace` to the signup flow, as part of 4326-gh-Automattic/p2
* Fixes CSS issues listed in 5136-gh-Automattic/p2

#### Testing instructions

`ENABLE_FEATURES=no-force-sympathy yarn start` on this branch.



1. Prepare the test P2 workspace. You will need to be admin to do this.
* Go to `10minutemail.com`, or similar temp mail providers. Grab the domain part of the temp email provided.
* Go to `https://wordpress.com/settings/general/YOUR-TEST-P2-WORKSPACE.wordpress.com` and add the domain as a pre-approved domain under the "Joining this workspace" section.

2. Go to `calypso.localhost:3000/start/p2` and sign up for an account using the temp email in Step 1. If on Chrome incognito, please turn off "Block third-party cookies" for testing convenience. 
3. Note: When you get redirected after clicking the confirmation email link, please make sure that you change `wordpress.com` back to `calypso.localhost:3000` in the URL.
4. After the "Complete your profile" step, you should see the new step. 
5. You should see your test P2 workspace listed as a workspace that the new user can join.
6. Verify that the code input form works as expected and you can join the workspace successfully.



<img width="800" alt="Screen Shot 2022-04-20 at 6 15 19 PM" src="https://user-images.githubusercontent.com/730823/164206285-6da41868-67eb-4b8d-b913-700fa99d37c9.png">

<img width="800" alt="Screen Shot 2022-04-21 at 12 42 12 AM" src="https://user-images.githubusercontent.com/730823/164281076-2626202e-48f7-4f5f-83cb-49e461f3b983.png">
